examples/mongo_odm/demo.cpp

```diff
@@ -0,0 +1,280 @@
+    return sysclock_t::time_point(since_epoch);
+}
+
+int main(int, char**) {
+    mongocxx::instance inst{};
+    mongocxx::client conn{mongocxx::uri{}};
+    auto db = conn["test"];
+    auto restaurants_col = db["restaurants"];
+    restaurants_col.delete_many({});
+    // wrap mongocxx::collection in an odm_collection
+    mongo_odm::odm_collection<Restaurant> restaurants(restaurants_col);
+
+    // Create sample restaurant object
+    Restaurant r1;
+    r1.address = {"1480", "2 Avenue", "10075", -73.9557413, 40.7720266};
+    r1.borough = "Manhattan";
+    r1.cuisine = "Italian";
+    r1.grades.push_back({sysclock_t::now(), "A", 11});
+    r1.grades.push_back({sysclock_t::now(), "B", 17});
+    r1.name = "Vella";
+    r1.restaurant_id = "41704620";
+
+    // Create sample restaurant object
+    Restaurant r2;
+    r2.address = {"558", "7 Avenue", "10018", -73.984472, 40.759011};
+    r2.borough = "Manhattan";
+    r2.cuisine = "Middle Eastern";
+    r2.grades.push_back({sysclock_t::now(), "B", 11});
+    r2.grades.push_back({sysclock_t::now(), "C", 17});
+    r2.name = "Maoz";
+    r2.restaurant_id = "41704621";
+
+    // // insert restaurants into collection
+    auto res = restaurants.insert_one(r1);
+    res = restaurants.insert_one(r2);
+
+    // get all restaurants in collection
+    {
+        std::cout << "Finding all restaurants in the collection..." << std::endl;
+        mongo_odm::deserializing_cursor<Restaurant> cur = restaurants.find({});
+        for (const auto& r : cur) {
+            std::cout << "Restaurant: " << r.name << std::endl;
+        }
+        std::cout << std::endl;
+    }
+
+    // Query by top-level field
+    {
+        std::cout << "Querying by cuisine (top-level field):" << std::endl;
+        bsoncxx::builder::stream::document filter;
+        filter << "cuisine"
+               << "Italian";
+        auto r_italian = restaurants.find_one(filter.view());
+        if (r_italian) {
+            std::cout << "Restaurant: " << r_italian->name << " (" << r_italian->cuisine << ")"
+                      << std::endl;
+        } else {
+            std::cout << "No matching restaurants were found." << std::endl;
+        }
+        std::cout << std::endl;
+    }
+
+    // Query by embedded field
+    {
+        std::cout << "Querying by street (embedded field):" << std::endl;
+        bsoncxx::builder::stream::document filter;
+        filter << "address.street"
+               << "7 Avenue";
+        auto cur = restaurants.find(filter.view());
+        for (const auto& r : cur) {
+            std::cout << "Restaurant: " << r.name << " " << r.address << std::endl;
+        }
+        std::cout << std::endl;
+    }
+
+    // Pass options to find()
+    {
+        std::cout << "Pass options to find(): " << std::endl;
+        mongocxx::options::find opts;
+        bsoncxx::builder::stream::document order_builder;
+        order_builder << "borough" << -1 << "address.zipcode" << 1;
```

Comment:
That's a bug with the C++ driver. https://docs.mongodb.com/manual/reference/limits/#Restrictions-on-Field-Names